feat(client): ✨ Signin

zaverbac opened the pull request.

zaverbac committed:
feat: ✨ Add google sign-in feature

Add user capability to sign-in using google auth

zaverbac committed:
feat: ✨ Add anonymous and microsoft signin

Add microsoft and anonymous signin by clicking on designated buttons

zaverbac committed:
fix: 🐛 Allow user to be added to emulator authenticator

Allow user to be added to emulators authenticator instead of dev authenticator

nstringham reviewed (changes requested):
Can you please remove all the unused images

nstringham reviewed (changes requested):
Please read over the linting errors and understand what they are saying and fix them if appropriate

nstringham commented:
it might be a good idea to make a file for auth instead of putting it all in the firebase file

zaverbac committed:
refactor(client): 🎨 Move auth function to seperate file, fix eslint errors, remove unsused images

Move auth functions to seperate file called authfunctions and fixed eslint errors and removed all unsused images

zaverbac commented:
Completed all suggestions in review

nstringham commented on client/src/lib/firebase/authfunctions.ts:
```diff
@@ -0,0 +1,42 @@
+export async function loginWithGoogle() {
+  const auth = getAuth(app);
+  const provider = new GoogleAuthProvider();
+  signInWithPopup(auth, provider).then((result) => {
+    // This gives you a Google Access Token. You can use it to access the Google API.
+    const credential = GoogleAuthProvider.credentialFromResult(result);
+    const token = credential!.accessToken;
+    // The signed-in user info.
+    const user = result.user;
+  });
```
it looks like we are not using any of these values so we probobly don't need the then block

nstringham commented on client/src/lib/firebase/authfunctions.ts:
```diff
@@ -0,0 +1,42 @@
+export async function loginWithGoogle() {
+  const auth = getAuth(app);
+  const provider = new GoogleAuthProvider();
+  signInWithPopup(auth, provider).then((result) => {
```
`signInWithPopup` is asyncronous so it should be awaited or returned

nstringham commented on client/src/lib/firebase/authfunctions.ts:
```diff
@@ -0,0 +1,42 @@
+  const auth = getAuth();
+  const provider = new OAuthProvider("microsoft.com");
+
+  signInWithPopup(auth, provider);
```
`signInWithPopup` is asyncronous so it should be awaited or returned

nstringham commented on client/src/lib/firebase/authfunctions.ts:
```diff
@@ -0,0 +1,42 @@
+// Anonymous login
+export function loginAnonymously() {
+  const auth = getAuth();
+  signInAnonymously(auth);
```
`signInAnonymously` is asyncronous so it should be awaited or returned

zaverbac committed:
refactor(client): ♻️ Add await to authFunctions  and add authstore file

Add await to authFunctions and displays promise to console. Add authstore file to hold store functions. Centered buttons in signin page

nstringham commented:
related to #27

zaverbac committed:
refactor(client): ♻️ Add await to auth functions

Remove eslint errors, remove unused variables, resolve eslint errors

nstringham committed:
refactor(client): ♻️ remove export of firebase app

because it is not used anywhere

nstringham committed:
refactor(client): 🚚 rename auth.ts

nstringham committed:
fix(client): 🚑️ update import with new name for auth.ts

this was broken by me in 1f4e94d696cc6ec869a64ebadd4409a012b6353d

nstringham commented on client/src/lib/stores/authstore.ts:
```diff
@@ -0,0 +1,12 @@
+import { writable, readable } from "svelte/store";
+
+export function getID() {
+  const { subscribe, set, update } = writable(localStorage.getItem("name"));
+
+  return {
+    subscribe,
+  };
+}
+if (typeof localStorage !== "undefined") {
+  getID();
+}
```
I think we should remove this in favor of rxfire

nstringham reviewed (changes requested):
Once you remove `authstore.ts` it should be good enough to merge and then we can make it better later

nstringham commented on client/src/routes/signin/+page.svelte:
```diff
@@ -0,0 +1,73 @@
+<!-- <h1>{$id}</h1> -->
+<style>
+  .container {
+    display: grid;
+    grid-template-columns: 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr;
+    grid-template-rows: 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr;
+    gap: 0px 0px;
+    grid-auto-flow: row;
+    grid-template-areas:
+      ". . . . . . . . ."
+      ". . . . . . . . ."
+      ". . . . . . . . ."
+      ". . . . google . . . ."
+      ". . . . microsoft . . . ."
+      ". . . . email . . . ."
+      ". . . . input . . . ."
+      ". . . . . . . . ."
+      ". . . . . . . . ."
+      ". . . . . . . . .";
```
If you are trying to center you should consider using `place-content`
https://css-tricks.com/snippets/css/complete-guide-grid/#aa-place-content

zaverbac committed:
refactor(client): 🔥 Remove authstore.ts and folder it is contained in

Delete authstore folder and file

nstringham reviewed (approved):
Good enough for now

zaverbac committed:
Merge branch 'development' of https://github.com/cat-acronym-team/cat-me-if-you-can into signin-feature
